fuel-vm/src/interpreter/executors/main/tests.rs

```diff
@@ -247,3 +247,51 @@ fn upgrade__tx_with_wrong_gas_price_causes_error() {
         InterpreterError::ReadyTransactionWrongGasPrice { .. }
     ));
 }
+
+#[test]
+fn upload__tx_with_wrong_gas_price_causes_error() {
+    // given
+    let tx_gas_price = 1;
+    let interpreter_gas_price = 2;
+    let input_amount = 1000;
+    let arb_max_fee = input_amount;
+    let consensus_params = ConsensusParameters::standard();
+
+    let interpreter_params = InterpreterParams {
+        gas_price: interpreter_gas_price,
+        ..Default::default()
+    };
+
+    let parts =
+        UploadPart::split_bytecode(&vec![123; 1024], 24).expect("Should split bytecode");
+    let part = parts[0].clone();
+    let ready_tx = TransactionBuilder::upload(UploadBody {
```

Comment:
Maybe move this to a small helper called `upload_tx_with_wrong_gas_price`. That's all the test reader needs to know about this `ready_tx` (which could be renamed to `bad_tx` or something like that. The `parts` could even go with it.

This will declutter the test a lot, shortening it by like 20 lines.